libr/core/cmd_open.c

```diff
@@ -1216,6 +1216,33 @@ static bool desc_list_json_cb(void *user, void *data, ut32 id) {
 	return true;
 }
 
+static bool cmd_op(RCore *core, char mode) {
+	int cur_fd = -2;
+	if (mode == 0) {
+		cur_fd = -1;
+		mode = 'n';
+	} else if (core->io && core->io->desc) {
+		cur_fd = core->io->desc->fd;
+	}
+	int limit = 1024;
+	int inc = (mode == 'n')? 1: -1;
+	if (cur_fd != -2) {
+		size_t i;
+		int fdn = cur_fd + inc;
+		for (i = 0; i < limit && fdn >= 0; i++) {
```

Comment:
instead of this, it's better if you add a `r_io_get_fds` in RIO. It should be easy to implement, because you can probably just use `r_id_storage_list (io->files)`.

Reply:
ive done the logc inside cmdopen.c without adding new apis, it doesnt looks that bad, but still this cmd_op can have a better name and get better arguments

Reply:
see the new implementation